crates/bevy_tasks/src/task_pool.rs

```diff
@@ -611,6 +532,10 @@ pub struct Scope<'scope, 'env: 'scope, T> {
 }
 
 impl<'scope, 'env, T: Send + 'scope> Scope<'scope, 'env, T> {
+    pub fn spawn(&self, f: impl FnOnce() + Send + 'scope) {
+        self.thread_pool.spawn(|_| f());
```

Comment:
This isn't quite right as the scope async executors can end while there are still tasks on the rayon scope. We need to push something into self.spawned. An easy fix here is to use spawn_async.